### PR TITLE
docs: update CLA information

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fix https://github.com/eslint/eslint/issues/15627

#### Is there anything you'd like reviewers to focus on?
nothing.

### CODE REVIEW

**Review by mdjermanovic — COMMENTED** [review]
The changes in `contributing/pull-requests.md` LGTM.

Is there anything we should update in the other document mentioned in https://github.com/eslint/eslint/issues/15627 ? (`contributing/README.md`)

**Comment by snitin315** [comment]
Not sure what should be updated there. @nzakas maybe you can help? 
https://github.com/eslint/eslint/blob/b93af98b3c417225a027cabc964c38e779adb945/docs/developer-guide/contributing/README.md?plain=1#L11-L14

**Comment by nzakas** [comment]
There isn't a direct link to the CLA anymore, so we should just remove the links and keep the text.

File: docs/developer-guide/contributing/README.md [hunk 1]
```diff
@@ -39,3 +35,7 @@ Have some extra time and want to contribute? This section talks about the proces
 ## Submitting a [Pull Request](pull-requests.md)
 
 We're always looking for contributions from the community. This section explains the requirements for pull requests and the process of contributing code.
+
+## Signing the CLA
+
+In order to submit code or documentation to an ESLint project, you will need to electronically sign our Contributor License Agreement. The CLA is the commonly used Apache-style template, and is you giving us permission to use your contribution. You only need to sign the CLA once for any OpenJS Foundation projects that use EasyCLA.
```
Comment (nzakas) [inline docs/developer-guide/contributing/README.md:41]:
```suggestion
In order to submit code or documentation to an ESLint project, you will need to electronically sign our Contributor License Agreement. The CLA is the commonly used Apache-style template, and is you giving us permission to use your contribution. You only need to sign the CLA once for any OpenJS Foundation projects that use EasyCLA. You will be asked to sign the CLA in the first pull request you open. 
```

**Review by nzakas — APPROVED** [review]

File: docs/developer-guide/contributing/README.md [hunk 2]
```diff
@@ -39,3 +35,7 @@ Have some extra time and want to contribute? This section talks about the proces
 ## Submitting a [Pull Request](pull-requests.md)
 
 We're always looking for contributions from the community. This section explains the requirements for pull requests and the process of contributing code.
+
+## Signing the CLA
+
+In order to submit code or documentation to an ESLint project, you will need to electronically sign our Contributor License Agreement. The CLA is the commonly used Apache-style template, and is you giving us permission to use your contribution. You only need to sign the CLA once for any OpenJS Foundation projects that use EasyCLA. You will be asked to sign the CLA in the first pull request you open.
```
Comment (nzakas) [inline docs/developer-guide/contributing/README.md:41]:
```suggestion
In order to submit code or documentation to an ESLint project, you will need to electronically sign our Contributor License Agreement. The CLA is the commonly used Apache-style template, and is you giving us permission to use your contribution. You only need to sign the CLA once for any OpenJS Foundation projects that use EasyCLA. You will be asked to sign the CLA in the first pull request you open.
```

**Review by nzakas — APPROVED** [review]

**Pull request merged**